runtime/j9vm/jvm.c

```diff
@@ -3998,6 +3998,44 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 }
 
 
+#if defined(J9VM_OPT_JAVA_OFFLOAD_SUPPORT) && (JAVA_SPEC_VERSION >= 17)
+/**
+ * Validates the JNI library for offload.
+ * If successful, returns a library handle which may be modified from the handle passed in,
+ * otherwise returns NULL.
+ *
+ * @param libName a null terminated string containing the library name.
+ * @param handle the handle of a JNI library using the libName.
+ *
+ * @return the shared library's handle if successful, which may be modified, NULL if out of memory.
+ *
+ * DLL: jvm
+ */
+void * JNICALL
+JVM_ValidateJNILibrary(const char *libName, void *handle)
+{
+	void *result = NULL;
+	UDATA doSwitching = 0;
+	PORT_ACCESS_FROM_JAVAVM(BFUjavaVM);
+
+	Trc_SC_ValidateJNILibrary_Entry(libName, handle);
+
+	/* If the library already has switching, just return it. */
+	if (J9_ARE_ANY_BITS_SET((UDATA)handle, J9_OFFLOAD_SWITCHING_FUNCTION_MASK)) {
+		return handle;
+	}
```

Comment:
I see two problems here:
1) this bypasses the exit tracepoint
1) this is an illegal call if any any of the mask bits are set; either the handle has already been "validated" or the underlying handle has those bits set in which case we can't support that handle